app/javascript/components/lists/ListGroup.js

```diff
@@ -0,0 +1,48 @@
+import React, { useState, useEffect } from "react"
+import ListGroupButtons from "./ListGroupButtons"
+
+const ListGroup = (props) => {
+  const [lists, setLists] = useState(props.lists);
+
+  // var lists = props.lists;
+  const csrf = props.csrf;
+
+  const handleStateChange = ( listId ) => {
+    var newLists = lists;
```

Comment:
This is the source of your bug.  `newLists` becomes a reference to `lists` instead of a copy of `lists`, which means when you call `splice` on `newLists` on L16, you're modifying the `lists`.  This causes a problem, because `setLists` will then look to see if there's a difference between what you passed (`newLists`) and what the current state is (`lists`), see no difference, and it will not detect a change, so it won't re-render.  You need to make this a shallow copy via the Spread Operator.  Doing so will allow you to remove the `window.location.reload()` on L21.

```suggestion
    var newLists = [...lists];
```

Reply:
Great catch thank you!